src/services/DebtService.sol

```diff
@@ -184,6 +164,28 @@ contract DebtService is AdminService {
         IERC20Permit(D_TOKEN).permit(msg.sender, address(this), _dAmt, _deadline, _v, _r, _s);
 
         // 2. Repay
-        repayAfterClose(_dAmt, _withdrawBuffer);
+        repayAndWithdraw(_dAmt, _withdrawBuffer);
+    }
+
+    /**
+     * @notice Calculates the maximum amount of collateral that can be withdrawn.
+     * @param _buffer The amount of collateral, in USD, left as safety buffer for tx to go through (at least 100_000 recommended, units: 8 decimals).
+     * @return maxWithdrawAmt The maximum amount of collateral that can be withdrawn (units: C_DECIMALS).
+     * uint256 cNeededUSD = (dTotalUSD * 1e4) / liqThreshold;
+     * uint256 maxWithdrawUSD = cTotalUSD - cNeededUSD - _buffer; (units: 8 decimals)
+     * maxWithdrawAmt = (maxWithdrawUSD * 10 ** (C_DECIMALS)) / cPriceUSD; (units: C_DECIMALS)
+     * Docs: https://docs.aave.com/developers/guides/liquidations#how-is-health-factor-calculated
+     */
+    function getMaxWithdrawAmt(uint256 _buffer) public view returns (uint256 maxWithdrawAmt) {
```

Comment:
Since this is now public, it the parameter name should be the same across all public functions (between this and close). Rename to `_withdrawBuffer`.